app/data_model/questionnaire_store.py

```diff
@@ -36,7 +35,7 @@ def __init__(self, user_id, user_ik):
             if 'COMPLETED_BLOCKS' in data:
                 self.completed_blocks = data['COMPLETED_BLOCKS']
 
-        self.initial_hash = hash(self.get_json())
+        self.data_on_load = self.get_json()
```

Comment:
In addition, since what this is doing is serialising and de-serialising the database raw data, create two functions with those names and do the json.loads and dumps in there (and merge in the get_data call?)